src/op_analytics/coreutils/partitioned/dailydata.py

```
@@ -102,6 +102,34 @@ def read(
         print(duckdb_context.client.sql("SHOW TABLES"))
         return view_name
 
+    @classmethod
+    def infer_all_schemas(cls, datestr: str) -> dict[str, dict]:
+        """Infer Clickhouse schemas for all datasets and return with metadata.
+
+        Args:
+            datestr: Date string to use for schema inference
+
+        Returns:
+            dict[str, dict]: Mapping of dataset names to their inferred schemas
+        """
+        schemas = {}
+        for table in cls.all_tables():
+            name = table.name
+            value = table.value.split(".")[-1]
+
+            log.info(f"Inferring schema for {name}")
+            try:
+                schema = table.infer_clickhouse_schema(datestr)
+                schemas[value] = schema
+            except IndexError:
+                log.warning(f"No data found for {name} on {datestr}, skipping schema inference")
+                continue
+            except Exception as e:
+                log.error(f"Error inferring schema for {name}", error=str(e))
+                continue
```

maybe on the docstring, could you add more info about where we would use the `infer_all_schemas` method? I want to understand where it would be used so I can think about the implications of ignoring missing dates or ignoring errors. 

Ate the very least I think we should never eat out an Exception, as we do in line 127.  Maybe the `IndexError` is ok to ignore.